src/javascripts/cookie-functions.js

```diff
@@ -115,7 +117,13 @@ export function setConsentCookie (options) {
     }
   }
 
+  // Set consent cookie
   setCookie(CONSENT_COOKIE_NAME, JSON.stringify(cookieConsent), { days: 365 })
+
+  // Initialise analytics if allowed
+  if (cookieConsent['analytics']) {
+    Analytics()
+  }
```

Comment:
As written I think this could end up starting the analytics twice... is that okay?

Reply:
I'm imagining a future scenario where we have an additional cookie category (not essential or analytics), and the user has opted-in to analytics, but changes their preference for the new cookie category.

Reply:
Thanks @lfdebrux , good point. I've tweaked a couple of things to help:

- https://github.com/alphagov/govuk-design-system/pull/1704/commits/0b4730623f3c3d0f25617c430240caa0b38a7fe5 new commit which adds a check for `window.ga.loaded` before the loadAnalytics script is run
- https://github.com/alphagov/govuk-design-system/pull/1704/commits/1240dd07c97b86de55d661eb1e15e419d0a8bd64 moved placement of this code to within the loop which goes through the `options` passed in `setConsentCookie`, to prevent the analytics script being run when other categories are changed in the consent cookie